src/lib/mdns/tests/TestDiscoveryManager.cpp

```diff
@@ -0,0 +1,81 @@
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+#include <nlunit-test.h>
+
+#include "lib/mdns/DiscoveryManager.h"
+#include "lib/support/CHIPMem.h"
+#include "platform/CHIPDeviceLayer.h"
+
+namespace {
+constexpr uint64_t kTestNodeId = 12344321;
+}
+
+class TestResolveDelegate : public chip::Mdns::ResolveDelegate
+{
+public:
+    TestResolveDelegate(nlTestSuite * suite) : mSuite(suite), mCount(0) {}
+
+    void HandleNodeIdResolve(CHIP_ERROR error, uint64_t nodeId, const chip::Mdns::MdnsService & address) override
+    {
+        NL_TEST_ASSERT(mSuite, error == CHIP_NO_ERROR);
+        NL_TEST_ASSERT(mSuite, nodeId == kTestNodeId);
+        NL_TEST_ASSERT(mSuite, address.mAddress.HasValue());
+        if (mCount == 0)
+        {
+            mCount++;
+            chip::Mdns::DiscoveryManager::GetInstance().ResolveNodeId(kTestNodeId, 0);
+        }
+        else
+        {
+            exit(0);
```

Comment:
This seems extreme and if hit this would 'retrurn tests pass'.

Please add a comment on why this is appropriate. Our automated scripts return 'success' on 0 exit codes, so this can hide away test failures.

Reply:
This is run inside an Thread since it requires the event loop to be working. Comment added.

Reply:
Could we thread join instead and let main return 0?

Reply:
Currently we lack mechanism to inform the chip event loop to gracefully exit so we can't.